Public/Src/FrontEnd/Script/Evaluator/Converter.cs

```diff
@@ -665,6 +665,57 @@ public static string ExtractStringLiteral(ObjectLiteral literal, SymbolAtom prop
             return ExpectNumber(value, context: new ConversionContext(name: property, objectCtx: literal));
         }
 
+        /// <summary>
+        /// Extracts optional boolean or int property.
+        /// Returns 'false' if the property is missing in a given object literal.
+        /// </summary>
+        public static bool ExtractOptionalBooleanOrInt(ObjectLiteral literal, SymbolAtom property, out bool enabled, out int intValue)
+        {
+            enabled = false;
+            intValue = 0;
+
+            var value = literal[property];
+            if (value.IsUndefined)
+            {
+                return false;
+            }
+
+            if(!TryGetBooleanOrInt(value, out enabled, out intValue))
+            {
+                throw CreateException(
+                   new[] { typeof(bool), typeof(int) },
+                   value,
+                   context: new ConversionContext(name: property, objectCtx: literal));
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Try to get int or bool value from EvaluationResult.
+        /// Returns 'false' if the value is no int or bool type.

```

Comment:
>'false' [](start = 20, length = 7)

If 'true', what's the right type, `bool` or `int`?

Reply:
It depends. It can be int or bool.